### PR TITLE
Fix move and rotation map tools preview when layer crs != canvas crs

### PR DESCRIPTION
**Pull request by YoannQDQ**
## Description

Fixes the Rotate Feature and Move Feature maptools previews, when the layer crs is not the same as the map canvas destination crs.

### Current behavior
broken_rubber

### Fixed behavior
fix_rubber

### CODE REVIEW

**Comment by roya0045** [comment]
Isn't it expensive to do a feature request and iteration on each update? If it's negligible or there's no way around, fine. Otherwise I would assume that creating a single entity and just moving it around would be more efficient, instead of refetching the data and transforming it?

**Comment by YoannQDQ** [comment]
You're probably right. I had no visible performance issue with my data, but they were extrememely simple polygons. I'll update the PR when I have the time.

File: src/app/qgsmaptoolmovefeature.cpp [hunk 1]
```diff
@@ -42,34 +42,55 @@ QgsMapToolMoveFeature::QgsMapToolMoveFeature( QgsMapCanvas *canvas, MoveMode mod
...
 void QgsMapToolMoveFeature::cadCanvasMoveEvent( QgsMapMouseEvent *e )
 {
   if ( mRubberBand )
   {
-    const QgsPointXY pointCanvasCoords = e->mapPoint();
-    const double offsetX = pointCanvasCoords.x() - mStartPointMapCoords.x();
-    const double offsetY = pointCanvasCoords.y() - mStartPointMapCoords.y();
-    mRubberBand->setTranslationOffset( offsetX, offsetY );
-    mRubberBand->updatePosition();
-    mRubberBand->update();
```
Comment (3nids) [inline src/app/qgsmaptoolmovefeature.cpp:57]: why are these 2 lines dropped?

Comment (YoannQDQ) [reply src/app/qgsmaptoolmovefeature.cpp:57]: If I'm not mistaken, `setTranslationOffset` already updates the rubburband, making the call to `updatePosition` and `update` redundant.

**Comment by YoannQDQ** [comment]
@3nids Should it be the other way around though? When drawing circles for instance (#51931), what matters is the shape be a circle in the project CRS, not in the layer CRS. So we would have to keep the rubberband as it was, and reproject the rubber geometry to the layer crs on release.

**Review by 3nids — APPROVED** [review]

**Pull request merged**